### PR TITLE
Filmic defaults update & cleanup

### PR DESCRIPTION
**Pull request by flannelhead**
As discussed in https://github.com/darktable-org/darktable/issues/12847:

* Change default norm to power norm
* Add checkbox to toggle the highlights reconstruction.
  - This allows turning it completely off which wasn't previously possible
  - New default is off, which works around https://github.com/darktable-org/darktable/issues/12442
  - When it's off, the sliders on that tab are disabled except for threshold and transition
  - When either of those sliders is slid, the reconstruction is automatically enabled.

Closes https://github.com/darktable-org/darktable/issues/12847.

As suggested in https://github.com/darktable-org/darktable/issues/12365#issuecomment-1312195504, remove the legacy Euclidean norm from the combobox by default. I hesitate doing the change of order proposed in https://github.com/darktable-org/darktable/issues/12365 this late in the dev cycle. Would be better to do it early in 4.3 cycle to gather feedback.

I tested these briefly, but more testing would be required. I won't be able to work on this PR tomorrow, so if any changes are desired and this PR is desired for 4.2, it would be better if someone else is able to contribute the changes.

### CODE REVIEW

**Review by TurboGit — COMMENTED** [review]
Just a thought, the threshold was put to +3EV to "disable" the highlight. Now that we have a checkbox wouldn't it better to set the threshold default to -2EV so when enabling reconstruction we have a default doing at least something?

File: src/iop/filmicrgb.c [hunk 1]
```diff
@@ -443,7 +447,7 @@ static inline void convert_to_spline_v3(dt_iop_filmicrgb_params_t* n)
 int legacy_params(dt_iop_module_t *self, const void *const old_params, const int old_version, void *new_params,
                   const int new_version)
 {
-  if(old_version == 1 && new_version == 5)
+  if(old_version == 1 && new_version == 6)
```
Comment (TurboGit) [inline src/iop/filmicrgb.c:450]: in each section we need to explicitly add:

`enable_highlight_reconstruction = TRUE;`

**Review by TurboGit — CHANGES_REQUESTED** [review]

**Comment by TurboGit** [comment]
Closing in favor of #12887.

**Comment by TurboGit** [comment]
> Just a thought, the threshold was put to +3EV to "disable" the highlight. Now that we have a checkbox wouldn't it better to set the threshold default to -2EV so when enabling reconstruction we have a default doing at least something?

Done in #12887.

**Comment by jorismak** [comment]
> Just a thought, the threshold was put to +3EV to "disable" the highlight. Now that we have a checkbox wouldn't it better to set the threshold default to -2EV so when enabling reconstruction we have a default doing at least something?

But isn't something like -0.10 EV a better default? -2 sounds very aggressive?

**Comment by TurboGit** [comment]
Not sure, I have always used a value between -3EV and -2EV to get proper recovering.

**Comment by Nilvus** [comment]
On my sound, setting lower than 0 is too hard. Just a bad setting. I agree with @jorismak

**Pull request closed without merging**